Created --reset-key flag to update OpenAI key

Ismail-Ben opened the pull request:
Resolves: #30 

Added --reset-key flag to update OPENAI_API_KEY in .sgptrc file. Also verified with a dummy chat completion that the newly inputted key works.

Sample output with 2 non-valid keys followed by a real key:
![image](https://github.com/TheR1D/shell_gpt/assets/48138437/87fdfb75-da73-43fc-a68c-b7abc24b44ae)

Ismail-Ben commented:
@TheR1D Any suggested changes to get this in?

TheR1D reviewed (changes requested):
I think this option doesn't align with prioritized environment variables in config now, since we can't reset user exported `OPENAI_API_KEY`. But we could add better error handling into `client._request` simply checking respone status code for `403/401` printing the error text and raising `Exit` with error `raise typer.Exit(code=1)`.

Ismail-Ben commented:
> I think this option doesn't align with prioritized environment variables in config now, since we can't reset user exported `OPENAI_API_KEY`. But we could add better error handling into `client._request` simply checking respone status code for `403/401` printing the error text and raising `Exit` with error `raise typer.Exit(code=1)`.

I've moved the error handling to client._request as suggested. 

As for the reset functionality I added a note to the "help" section of the command as a reminder that the environment variable will still be prioritized. I still believe having a --reset-key option is worthwhile as I would've liked to use it myself when first trying the tool. Instead of having to navigate to the .sgptrc file and update it manually I would've been nice to have this command. 

The config.get() function properly prioritizes the use of the env key if it exists so updating the .sgptrc file wouldn't change this behavior.  In the case that a user runs this command while already having the env variable I highlight in yellow that their env key will still be used.

User updates key with env variable defined:
![image](https://github.com/TheR1D/shell_gpt/assets/48138437/badde450-3e81-4ec2-af11-2b280e610f00)

No env variable defined, and fake key:
![image](https://github.com/TheR1D/shell_gpt/assets/48138437/c78ba71b-4bc7-45f3-b044-dbb19fc589f0)

Help section:
![image](https://github.com/TheR1D/shell_gpt/assets/48138437/ac241758-5781-468c-8676-098f058c9c88)

Let me know what you think @TheR1D . Thanks!

TheR1D commented on sgpt/client.py:
```diff
@@ -58,7 +59,15 @@ def _request(
             stream=stream,
         )
-        response.raise_for_status()
+        try:
```
There is no reason to use `try` block here, we can just check `response.status` above.

TheR1D commented on sgpt/client.py:
```diff
@@ -58,7 +59,15 @@ def _request(
+        except requests.exceptions.HTTPError as http_err:
+            # Check if the status code is 403 (Forbidden) or 401 (Unauthorized)
+            if http_err.response.status_code == 403 or http_err.response.status_code == 401:
+                typer.secho("Invalid OpenAI API key.", fg="red")
```
We can add more info for user, something like `f"Invalid OpenAI API key, update your config file: {cfg.SHELL_GPT_CONFIG_PATH}"`.

TheR1D commented on sgpt/client.py:
```diff
@@ -58,7 +59,15 @@ def _request(
+            if http_err.response.status_code == 403 or http_err.response.status_code == 401:
+                typer.secho("Invalid OpenAI API key.", fg="red")
+            else:
+                typer.secho(f"HTTPError occurred: {http_err}", fg="red")
```
We don't need this line since `raise_for_status` should handle it.

TheR1D reviewed (changes requested):
It is very common in CLI applications to use runtime configuration file or ENV variables to set API keys (without having dedicated CLI options to reset some of config values). Please remove reset key option, if we show user proper message, it is clear how to reset it.

Ismail-Ben commented:
Adjusted to reflect @TheR1D 's comments. When using an invalid key the output now looks like:
![image](https://github.com/TheR1D/shell_gpt/assets/48138437/a93205c5-c336-4156-acaa-b28122c8c5d4)

TheR1D reviewed (dismissed):
Thank you @Ismail-Ben

TheR1D commented:
Please check lint issues.

TheR1D commented:
Thank you, merged.
